src/components/docs-version-switcher/helpers.ts

```diff
@@ -0,0 +1,57 @@
+/**
+ * Copied from @hashicorp/react-version-select on 2022-05-24
+ * @see https://github.com/hashicorp/react-components/blob/22a02a47f5dcac983e98f0f3096cf0a785447f7d/packages/version-select/util.ts
+ */
+
+const REGEX = /^v\d+\.\d+\.(\d+|\w+)$/i
+
+/**
+ * Extract the version from a path string, or an array of path segments
+ */
+function getVersionFromPath(path: string): string | undefined {
```

Comment:
🎨  I think there's a version of this in `lib/get-version-from-path`.

I wonder if should `export const REGEX = /^v\d+\.\d+\.(\d+|\w+)$/i` from there as well.... or perhaps move this entire file to `lib/get-version-from-path`? (Maybe rename to `lib/version-path-helpers` or something? 🤔 )

Reply:
Oh nice! These are the exact same functions. I didn't realize that we had already pulled (at least) one of them over. 😊 I'll go ahead and lift up the others as well where needed. Great catch!

Reply:
Addressed in:

- https://github.com/hashicorp/dev-portal/pull/491/commits/1299686bbd2812e08b07f1138dd3bb8d1c1ddf38
- https://github.com/hashicorp/dev-portal/pull/491/commits/a738a55cea7d821e7098607ef4f5060501e500d0